src/pages/ServiceGraph/SummaryPanelCommon.ts

```diff
@@ -29,11 +44,48 @@ export const updateHealth = (summaryTarget: any, stateSetter: (hs: HealthState)
   }
 };
 
-export const nodeData = (node: any) => {
+export const nodeData = (node: any): NodeData => {
   return {
     namespace: node.data('namespace'),
     app: node.data('app'),
     version: node.data('version'),
     workload: node.data('workload')
   };
 };
+
+export const nodeGlobalData = (node: any) => {
+  return node.cy().scratch(CytoscapeGlobalScratchNamespace) as CytoscapeGlobalScratchData;
+};
+
+export const getNodeMetrics = (
+  node: any,
+  props: SummaryPanelPropType,
+  filters: Array<string>,
+  includeIstio?: boolean,
+  byLabelsIn?: Array<string>
+) => {
+  // Determine if is by workload or by app
+  const data = nodeData(node);
+  const globalData = nodeGlobalData(node);
+  const isByApp = globalData.graphType === GraphType.APP && data.app !== 'unknown';
```

Comment:
Are you sure about this check? I have a similar check to perform for health, see on my PR: https://github.com/kiali/kiali-ui/pull/546/files#diff-04b15b4602d6b9ccf1eb82d15e07535fR485
If an app box is selected, in workload-based graph, I guess it should show app-based metrics.

Reply:
I think there isn't any "app box" in workload-based graph (at least not yet).
If there is plans for one I could update this check.

Reply:
Fair enough, this is still because I had wrong assumptions on how the graph was generated.